bsp/Src/CAN.c

```diff
@@ -190,156 +191,102 @@ static inline void HAL_CAN_MspF4Init(CAN_HandleTypeDef* hcan){
     PA11 -> CAN1_RX
     PA12 -> CAN1_TX
     */
-    init.Pin = GPIO_PIN_11;
+    init.Pin = GPIO_PIN_11 | GPIO_PIN_12;
     init.Mode = GPIO_MODE_AF_PP;
     init.Pull = GPIO_PULLUP;
     init.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
     init.Alternate = GPIO_AF9_CAN1;
     HAL_GPIO_Init(GPIOA, &init);
-
-    init.Pin = GPIO_PIN_12;
-    init.Mode = GPIO_MODE_AF_PP;
-    init.Pull = GPIO_NOPULL;
-    init.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
-    init.Alternate = GPIO_AF9_CAN1;
-    HAL_GPIO_Init(GPIOA, &init);
```

Comment:
pin 12 has no pullup here and pin 11 does; by combining them you're making them both pullup; is that what you want to do?